### ISSUE TITLE
Make database searchable

### ISSUE THREAD

**Issue by yurumi** [ifosch/keepass-mode#3]
Hi,

first of all, thanks for your efforts! I think usability would be greatly improved if one could search the database to find a specific entry. As my brain is not compatible with (e)lisp I'm afraid I cannot contribute though...

Cheers,
Thomas

**Comment by Gleek** [comment]
I've a small snippet in my config that helps me in doing this using `ivy`.
See if it works for you as well.
Here's a [permalink](https://github.com/Gleek/.emacs/blob/5d8872c4e2a0ddad8685734f5c3343d1a1df54b3/core/core-tools.el#L315-L360). (For the latest state refer to the [master branch](https://github.com/Gleek/.emacs/search?q=%22keepass-mode%22))
It helps you search in all your keys by pressing `s` if in keepass-mode. Or pressing `C-c s p` from anywhere in emacs, once you have added `(setq keepass-password-file "my-file-location.kbdx")` somewhere in your config.

**Comment by yurumi** [comment]
Thanks for sharing - it works like a charm!

Although this covers most use cases, I leave the issue open because it would be nice to have a full text search (also in the notes/username) not just in the keys...

**Comment by ifosch** [comment]
Hi!

Sorry for such delayed response. I've been busy with a lot of things lately. I am looking forward to improve the release process and the testing of this project. However, I think this feature is worth to take a look at.

I am considering to take a deeper look at @Gleek snippet to see if there is anything worth to actually include in the project's code.

**Comment by ykgoon** [comment]
I agree with need this feature. Full text search is the typical way even in GUI UX.

Navigating group tree is cumbersome.

**Comment by bhankas** [comment]
@Gleek 's function works very well. Except it requires ivy. Since keepass-cli is usually too fast to notice, we can even have interactive completion like with GUI KeepassXC.

**Comment by Gleek** [comment]
> Gleek 's function works very well. Except it requires ivy

@bhankas I've recently started to try out the vertico, consult, embark ecosystem and have ported this to work with consult with embark actions (along with the existing ivy integration) for copying password, username and opening the entry. Maybe you'll find it useful.
Here's the [link](https://github.com/Gleek/emacs.d/blob/master/packages/keepass-mode-plus.el)

PS: To use it with ivy doing `(setq +keepass-completion 'ivy)` should switch the completion framework.
